dotCMS/src/main/java/com/dotcms/rest/api/v1/announcements/AnnouncementsResource.java

```diff
@@ -0,0 +1,83 @@
+package com.dotcms.rest.api.v1.announcements;
+
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.system.announcements.Announcement;
+import com.fasterxml.jackson.jaxrs.json.annotation.JSONP;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * This REST Resource behaves as a proxy to the remote dotCMS instance that provides the announcements
+ */
+@Path("/v1/announcements")
+public class AnnouncementsResource {
+
+    private final WebResource webResource;
+
+    private final AnnouncementsHelper helper;
+
+    /**
+     * Default constructor
+     */
+    public AnnouncementsResource() {
+        this.webResource = new WebResource();
+        this.helper = new AnnouncementsHelperImpl();
+    }
+
+    /**
+     * Constructor
+     * @param webResource WebResource
+     * @param helper AnnouncementsHelper
+     */
+    @VisibleForTesting
+    public AnnouncementsResource(WebResource webResource, AnnouncementsHelper helper) {
+        this.webResource = webResource;
+        this.helper = helper;
+    }
+
+    /**
+     * Get the announcements
+     * @param request HttpServletRequest
+     * @param response HttpServletResponse
+     * @param langIdOrCode String
+     * @param refreshCache boolean
+     * @param limit int
+     * @return Response
+     */
+    @GET
+    @Path("/")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response announcements(@Context final HttpServletRequest request,
```

Comment:
we prefer to return a response entity view instead of Response to have better docs generated

Reply:
true